app/bg/dbs/settings.js

```diff
@@ -107,9 +119,12 @@ export const getAll = function () {
   return setupPromise.then(v => cbPromise(cb => {
     db.all(`SELECT key, value FROM settings`, (err, rows) => {
       if (err) { return cb(err) }
-
       var obj = {}
       rows.forEach(row => { obj[row.key] = row.value })
+      // parse non-string values
+      if (obj.default_search_engines) obj.default_search_engines = JSON.parse(obj.default_search_engines)
+      if (obj.selected_search_engine) obj.selected_search_engine = JSON.parse(obj.selected_search_engine)
```

Comment:
Make sure to use the `JSON_ENCODED_SETTINGS.includes(key)` logic here instead of hardcoding them.